src/EFCore.PG/Migrations/NpgsqlMigrationsModelDiffer.cs

```diff
@@ -0,0 +1,218 @@
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking.Internal;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Migrations.Internal;
+using Microsoft.EntityFrameworkCore.Migrations.Operations;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.EntityFrameworkCore.Update.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Migrations
+{
+    /// <summary>
+    /// The default migrations model differ for Npgsql.
+    /// </summary>
+    public class NpgsqlMigrationsModelDiffer : MigrationsModelDiffer
+    {
+        /// <inheritdoc />
+        public NpgsqlMigrationsModelDiffer(
+            [NotNull] IRelationalTypeMappingSource typeMappingSource,
+            [NotNull] IMigrationsAnnotationProvider migrationsAnnotations,
+            [NotNull] IChangeDetector changeDetector,
+            [NotNull] StateManagerDependencies stateManagerDependencies,
+            [NotNull] CommandBatchPreparerDependencies commandBatchPreparerDependencies)
+            : base(typeMappingSource, migrationsAnnotations, changeDetector, stateManagerDependencies, commandBatchPreparerDependencies) {}
+
+        [NotNull]
+        [ItemNotNull]
+        protected override IEnumerable<string> GetSchemas(IModel model)
+            => base.GetSchemas(model)
+                   .Concat(model.Npgsql().PostgresExtensions.Select(x => x.Schema))
+                   .Concat(model.Npgsql().PostgresEnums.Select(x => x.Schema))
+                   .Concat(model.Npgsql().PostgresRanges.Select(x => x.Schema))
+                   .Where(x => !string.IsNullOrEmpty(x))
+                   .Distinct();
+
+        protected override IEnumerable<MigrationOperation> Diff(IModel source, IModel target, DiffContext diffContext)
+            => base.Diff(source, target, diffContext)
+                   .Concat(DiffExtensions(source, target))
+                   .Concat(DiffEnums(source, target))
+                   .Concat(DiffRanges(source, target));
+
+        [NotNull]
+        [ItemNotNull]
+        protected virtual IEnumerable<MigrationOperation> DiffExtensions([CanBeNull] IModel source, [CanBeNull] IModel target)
+        {
+            var sourceExtensions = source?.Npgsql().PostgresExtensions;
+            var targetExtensions = target?.Npgsql().PostgresExtensions;
+
+            if (source == null && targetExtensions != null)
+            {
+                if (targetExtensions.Count == 0)
+                    yield break;
```

Comment:
This isn't strictly required, but it keep us from yielding back empty migration operations.